### PR TITLE
[local gc] Fix for BitScanForward64, it was ignoring the high byte check

### PR DESCRIPTION
**Pull request by davmason**
Fix for #17117. Our implementation of BitScanForward64 would check the high bits, then overwrite `bitIndex` on the call to BitScanForward for the low bits.

We never saw this in coreclr because the compiler intrinsic `_BitScanForward64` is used, and our implementation was only called in clrgc.dll. It's not obvious to me why that is, but this fixes the issue.

I ran the default set of tests with runtest.cmd with local gc enabled, and have a stress run going overnight.

### CODE REVIEW

File: src/gc/env/gcenv.base.h [hunk 1]
```diff
@@ -253,12 +253,21 @@ inline uint8_t BitScanForward64(uint32_t *bitIndex, uint64_t mask)
...
+    if (result != 0)
+    {
+        return result;
+    }
+
+    result = BitScanForward(&fakeBitIndex, hi);
+    if (result != 0)
     {
         *bitIndex = fakeBitIndex + 32;
+        return result;
```
Comment (janvorli) [inline src/gc/env/gcenv.base.h]: You can remove this 'return result' and replace the `return 0` at the end by `return result`. And maybe simplify it even more:
```C
uint8_t result = BitScanForward(bitIndex, lo);
if (result == 0)
{
    result = BitScanForward(&fakeBitIndex, hi);
    if (result != 0)
    {
        *bitIndex = fakeBitIndex + 32;
    }
}

return result;
```

Comment (davmason) [reply src/gc/env/gcenv.base.h]: Thanks, Jan

**Comment by davmason** [comment]
@dotnet-bot test Tizen armel Cross Checked Innerloop Build and Test

**Comment by davmason** [comment]
@dotnet-bot test Tizen armel Cross Checked Innerloop Build and Test

**Comment by davmason** [comment]
Per #17028, tizen job is misbehaving, going to merge now

**Pull request merged**